src/transformers/models/imagebind/feature_extraction_imagebind.py

```diff
@@ -0,0 +1,413 @@
+            The duration of each chunk in seconds.
+        num_chunks (`int`, *optional*, defaults to 3):
+            The number of chunks to sample from the input audio.
+    """
+
+    model_input_names = ["input_features"]
+
+    def __init__(
+        self,
+        feature_size=1,
+        sampling_rate=16000,
+        num_mel_bins=128,
+        max_length=204,
+        padding_value=0.0,
+        do_normalize=True,
+        mean=-4.268,
+        std=9.138,
+        do_chunk=True,
+        chunk_duration=2.0,
+        num_chunks=3,
+        **kwargs,
+    ):
+        super().__init__(feature_size=feature_size, sampling_rate=sampling_rate, padding_value=padding_value, **kwargs)
+        self.num_mel_bins = num_mel_bins
+        self.max_length = max_length
+        self.do_normalize = do_normalize
+        self.mean = mean
+        self.std = std
+        self.do_chunk = do_chunk
+        self.chunk_duration = chunk_duration
+        self.num_chunks = num_chunks
+
+        if not is_speech_available():
+            mel_filters = mel_filter_bank(
+                num_frequency_bins=256,
+                num_mel_filters=self.num_mel_bins,
+                min_frequency=20,
+                max_frequency=sampling_rate // 2,
+                sampling_rate=sampling_rate,
+                norm=None,
+                mel_scale="kaldi",
+                triangularize_in_mel_space=True,
+            )
+
+            self.mel_filters = np.pad(mel_filters, ((0, 1), (0, 0)))
+            self.window = window_function(400, "hann", periodic=False)
+
+    def _extract_fbank_features(
+        self,
+        waveform: np.ndarray,
+        max_length: int,
+    ) -> np.ndarray:
+        """
+        Get mel-filter bank features using TorchAudio. Note that TorchAudio requires 16-bit signed integers as inputs
+        and hence the waveform should not be normalized before feature extraction.
+        """
+        # waveform = waveform * (2**15)  # Kaldi compliance: 16-bit signed integers
+        # Mean center the waveform
+        waveform -= waveform.mean()
+
+        if is_speech_available():
+            waveform = torch.from_numpy(waveform).unsqueeze(0)
+            fbank = ta_kaldi.fbank(
+                waveform,
+                sample_frequency=self.sampling_rate,
+                window_type="hanning",
+                num_mel_bins=self.num_mel_bins,
+            )
+        else:
+            waveform = np.squeeze(waveform)
+            fbank = spectrogram(
+                waveform,
+                self.window,
+                frame_length=400,
+                hop_length=160,
+                fft_length=512,
+                power=2.0,
+                center=False,
+                preemphasis=0.97,
+                mel_filters=self.mel_filters,
+                log_mel="log",
+                mel_floor=1.192092955078125e-07,
+                remove_dc_offset=True,
+            ).T
+
+            fbank = torch.from_numpy(fbank)
```

Comment:
Since the model is using torch already, I don't mind keeping this dependent on torch

Reply:
But we could probably do the rest of the operation in numpy right ?

Reply:
I tried doing the rest of the operation in numpy like this:
```
        # Convert to [mel_bins, num_frames] shape
        fbank = fbank.transpose(0, 1)
        # Pad to max_length
        n_frames = fbank.shape[1]
        
        difference = max_length - n_frames

        if abs(difference) / n_frames > 0.2:
            logger.warning_once(
                f"Large padding or truncation for {tuple(waveform.shape)} waveform with {n_frames} frames and {max_length} max_length."
            )

        if difference > 0:
            # Padding with zeros
            fbank = np.pad(fbank, ((0, 0), (0, difference)), mode='constant', constant_values=0)
        elif difference < 0:
            fbank = fbank[:, :max_length]  # Truncate to max_length

        fbank = np.expand_dims(fbank, axis=0)
```
but keeping it in numpy during padding or truncation results in inconsistent shapes and fails the test

Reply:
Update: had to overwrite previous comment as the changes were wrong because numpy operations were performed on torch tensors

Reply:
> But we could probably do the rest of the operation in numpy right ?

@ylacombe When `if is_speech_available():` , `ta_kaldi.fbank` is used to perform operation on the numpy array converted into torch tensor like as you mentioned in next review  comment that  numpy to torch conversion should be done during `ta_kaldi.fbank`. So, the output is a torch tensor and that's why at `line 267` that you referred, numpy to torch conversion has to be done when `is_speech_available` is False because out of the `if-else` block padding operations are done with torch because when `is_speech_available` is True, we have output as torch tensor to proceed ahead